src/app/auth/account/account.component.ts

```diff
@@ -1,8 +1,190 @@
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class AccountComponent {}
+export class AccountComponent extends BaseComponent implements OnInit {
+    public accountViewType = AccountTabs.overview;
+    private _window: Window;
+    private resizeId: NodeJS.Timer;
+    private menuRowWidth: number;
+    public user: MaybeNull<AccountDetailsFragment>;
+    public datasets: DatasetSearchOverviewFragment[] = [];
+    public pageInfo: PageBasedInfo;
+    public accountTabs = AccountTabs;
+    public accountName: string;
+    public isClickableRow = true;
+    public isDropdownMenu = false;
+    public isCurrentUser = false;
+    public currentPage = 1;
+
+    @ViewChild("containerMenu") containerMenu: ElementRef;
+    @ViewChild("dropdownMenu") dropdownMenu: ElementRef;
+
+    constructor(
+        private authApi: AuthApi,
+        private route: ActivatedRoute,
+        private router: Router,
+        private navigationService: NavigationService,
+        private cdr: ChangeDetectorRef,
+        private modalService: ModalService,
+    ) {
+        super();
+        this._window = window;
+        this.user = this.authApi.currentUser;
+    }
+
+    public ngOnInit(): void {
+        this.trackSubscriptions(
+            this.route.queryParams.subscribe((param: Params) => {
+                if (param.tab) {
+                    this.accountViewType = param.tab as AccountTabs;
+                }
+                if (param.page) {
+                    this.currentPage = param.page as number;
+                }
+            }),
+            this.router.events
+                .pipe(
+                    filter((event) => event instanceof NavigationEnd),
+                    map((event) => event as RouterEvent),
+                )
+                .subscribe(() => this.getDatasets()),
+        );
+        if (this.user) {
+            this.accountName = this.user.login;
+            this.getDatasets();
+        }
+    }
+
+    public get isAccountViewTypeOverview(): boolean {
+        return this.accountViewType === AccountTabs.overview;
+    }
+
+    public get isAccountViewTypeDatasets(): boolean {
+        return this.accountViewType === AccountTabs.datasets;
+    }
+
+    public get isAccountViewTypeOrganizations(): boolean {
+        return this.accountViewType === AccountTabs.organizations;
+    }
+
+    public get isAccountViewTypeInbox(): boolean {
+        return this.accountViewType === AccountTabs.inbox;
+    }
+
+    public get isAccountViewTypeStars(): boolean {
+        return this.accountViewType === AccountTabs.stars;
+    }
+
+    public onEditProfile(): void {
+        promiseWithCatch(
+            this.modalService.warning({
+                message: AppValues.UNIMPLEMENTED_MESSAGE,
+                yesButtonText: "Ok",
+            }),
+        );
+    }
+
+    public selectedTabs(accountTabKey: AccountTabs): boolean {
+        if (this._window.location.search.includes(accountTabKey)) {
```

Comment:
Base component class has `searchString` helper. You will not need `_window` property if you use it instead.